### PR TITLE
feat(ramp on/off): add support for dark theme and typography

### PR DESCRIPTION
**Pull request by SorinC6**
task: https://emurgo.atlassian.net/browse/YOMO-1108

demo: https://recordit.co/teiTnEGuYP

Screenshots:

Screenshot 2024-01-17 at 15 17 33
Screenshot 2024-01-17 at 15 18 40

### CODE REVIEW

**Review by michalszorad — COMMENTED** [review]
Looks good, but seems like there are some build issues

File: apps/wallet-mobile/src/Search/SearchContext.tsx [hunk 1]
```diff
@@ -128,7 +130,7 @@ export const useSearchOnNavBar = ({
 
   const withSearchInput: StackNavigationOptions = React.useMemo(
     () => ({
-      ...defaultStackNavigationOptions,
+      ...defaultStackNavigationOptions(theme),
```
Comment (stackchain) [inline apps/wallet-mobile/src/Search/SearchContext.tsx]: hey @SorinC6 maybe collocate when there is plenty of references.
```

const {theme} = useTheme()
const defaultNavigationOptions = React.useCallback(() => defaultStackNavigationOptions(theme), [theme])

```

File: apps/wallet-mobile/src/TxHistory/TxHistoryNavigator.tsx [hunk 2]
```diff
@@ -234,7 +236,7 @@ export const TxHistoryNavigator = () => {
                   name="send-start-tx"
                   options={{
                     title: strings.sendTitle,
-                    ...sendOptions,
+                    ...sendOptions(theme),
```
Comment (stackchain) [inline apps/wallet-mobile/src/TxHistory/TxHistoryNavigator.tsx:245]: :arrow_up:

File: apps/wallet-mobile/src/components/StatusBar.tsx [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
   type: 'dark' | 'light'
 }
 
 export const StatusBar = ({type}: Props) => {
-  const backgroundColor = type === 'dark' ? COLORS.WHITE : COLORS.BACKGROUND_BLUE
+  const {colorScheme, theme} = useTheme()
+  const backgroundColor = type === 'dark' ? theme.color['white-static'] : theme.color['black-static']
 
-  return <NativeStatusBar style={type} backgroundColor={backgroundColor} />
+  return <NativeStatusBar style={colorScheme === 'dark' ? 'light' : 'dark'} backgroundColor={backgroundColor} />
```
Comment (stackchain) [inline apps/wallet-mobile/src/components/StatusBar.tsx]:
```suggestion
  return <NativeStatusBar style={isDark ? 'light' : 'dark'} backgroundColor={backgroundColor} />
```

File: apps/wallet-mobile/src/components/StatusBar.tsx [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
 
-import {COLORS} from '../theme'
-
 type Props = {
   type: 'dark' | 'light'
 }
 
 export const StatusBar = ({type}: Props) => {
-  const backgroundColor = type === 'dark' ? COLORS.WHITE : COLORS.BACKGROUND_BLUE
+  const {colorScheme, theme} = useTheme()
```
Comment (stackchain) [inline apps/wallet-mobile/src/components/StatusBar.tsx]:
```suggestion
  const {theme, isDark} = useTheme()
```

File: apps/wallet-mobile/src/features/RampOnOff/useCases/CreateExchange/ShowDisclaimer/ShowDisclaimer.tsx [hunk 5]
```diff
@@ -25,26 +26,29 @@ export const ShowDisclaimer = () => {
...
+    },
+    gradient: {
+      opacity: 1,
+      borderRadius: 8,
+      marginTop: 24,
+    },
+    title: {
+      ...theme.typography['body-1-regular'],
+      color: theme.color.gray.max,
+      fontWeight: '500',
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/RampOnOff/useCases/CreateExchange/ShowDisclaimer/ShowDisclaimer.tsx:44]: @SorinC6 can you check with Design, I really don't like the idea of having the Typography set, and still need to add `fontWeight` or whatever other property to the fonts (family/weight/space) that is not pointing to the typography directly.

File: apps/wallet-mobile/src/features/RampOnOff/useCases/ShowExchangeResult/ShowExchangeResult.tsx [hunk 6]
```diff
@@ -112,19 +114,15 @@ const useStyles = () => {
       height: 228,
     },
     congratsText: {
-      fontSize: 22,
+      ...theme.typography['heading-3-regular'],
       color: theme.color.gray[900],
-      fontFamily: 'Rubik',
-      lineHeight: 30,
       fontWeight: '500',
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/RampOnOff/useCases/ShowExchangeResult/ShowExchangeResult.tsx:119]: :arrow_up:

File: apps/wallet-mobile/src/navigation.tsx [hunk 7]
```diff
@@ -31,41 +32,47 @@ export const useParams = <Params, >(guard: Guard<Params>): Params => {
...
+export const defaultStackNavigationOptions = (theme: Theme): StackNavigationOptions => {
+  return {
+    headerTintColor: theme.color.gray.max,
+    headerStyle: {
+      elevation: 0,
+      shadowOpacity: 0,
+      backgroundColor: theme.color.gray.min,
+    },
+    headerTitleStyle: {
+      fontSize: 16,
```
Comment (stackchain) [inline apps/wallet-mobile/src/navigation.tsx]: is there a Typography for this one?

File: apps/wallet-mobile/src/features/RampOnOff/useCases/CreateExchange/ShowDisclaimer/ShowDisclaimer.tsx (hunk 5 above)
Comment (SorinC6) [reply apps/wallet-mobile/src/features/RampOnOff/useCases/CreateExchange/ShowDisclaimer/ShowDisclaimer.tsx:44]: @stackchain Have looked into it before but was not sure. To have the fontWeight in the theme we need to introduce something like this: `body-1-regular-500` ,  `body-1-regular-400` , and `body-1-regular-600`,  for most of the theme text typography variants

Comment (stackchain) [reply apps/wallet-mobile/src/features/RampOnOff/useCases/CreateExchange/ShowDisclaimer/ShowDisclaimer.tsx:44]: hmm, @SorinC6 my point is, `body-<1>` there is zero semantic meaning of this `1` what does it represent? a combination of what? mode + weight? only weight? only size? Let's arrange a call with Design, for me a good semantic would be `body-small-regular` + `strong`, this clearly indicates that fontWeight is not part of the first token, and I'd like to have a consistency around `small` `normal` `big`, we should leverage the fontScale, cuz some devices folks bump up a lot the their fonts, therefore, small, normal, big, whatever, should be a `base * fontScale * <small | normal | big>`

Comment (SorinC6) [reply apps/wallet-mobile/src/features/RampOnOff/useCases/CreateExchange/ShowDisclaimer/ShowDisclaimer.tsx:44]: @stackchain I've been following the pattern as h1,h2,h3... as on the design are\
  ```
'heading-1-regular': {
    ...fontFamily.regular,
    ...size.h1,
  },
  'heading-2-medium': {
    ...fontFamily.medium,
    ...size.h2,
  },
  'heading-2-regular': {
    ...fontFamily.regular,
    ...size.h2,
  },
  'heading-3-medium': {
    ...fontFamily.medium,
    ...size.h3,
  },
  'heading-3-regular': {
    ...fontFamily.regular,
    ...size.h3,
  },
```

And for the body adding the same pattern 🤔

```
  body1: {
    fontSize: 16,
    lineHeight: 24,
  },
  body2: {
    fontSize: 14,
    lineHeight: 22,
  },
  body3: {
    fontSize: 12,
    lineHeight: 18,
  },
```

Then adding this 
```

  'body-1-medium': {
    ...fontFamily.medium,
    ...size.body1,
  },
  'body-1-regular': {
    ...fontFamily.regular,
    ...size.body1,
  },
  'body-2-medium': {
    ...fontFamily.medium,
    ...size.body2,
  },
```

for the heading I think we can keep it as 1 ,2,3.. as above and the body we can change as: 
- body-small-medium-(fontWeight) , body-normal-medium-(fontWeight), body-big-medium-(fontWeight), body-small-regular-(fontWeight), body-normal-regular-(fontWeight), body-big-regular--(fontWeight)

**Comment by stackchain** [comment]
ephemeral state - bringing in for updates.

**Pull request merged**